java-checks/src/main/java/org/sonar/java/symexecengine/ExecutionState.java

```diff
@@ -36,18 +37,41 @@
 
 public class ExecutionState {
 
-  final ExecutionState parent;
-  private SetMultimap<Symbol, SymbolicValue> reachableValues = HashMultimap.create();
+  public final ExecutionState parent;
+
+  /*
+   * definitelyReachableValues and potentiallyReachableValues contains the set of possible values from a given symbol.
+   * - definitelyReachableValues contains the values that are definitely retrieved by a read operation.
+   * - potentiallyReachableValues contains all the values that can be retrieved by a read operation.
+   * this is illustrated in the examples below:
+   * example 1                      example 2                        example 3
+   * a = V1;                        a = V1;                          a = V1;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)